### PR TITLE
Refactor build srcs to separate file

### PR DESCRIPTION
**Pull request by mcr229**
I was hoping to refactor the src definitions in the bazel file out out of BUILD.bazel. This just helps make other build systems (BUCK) to more easily leverage the existing build definitions in XNNPACK.

#### Test
```
bazel build :XNNPACK         
INFO: Analyzed target //:XNNPACK (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //:XNNPACK up-to-date:
  bazel-bin/libXNNPACK.a
INFO: Elapsed time: 0.352s, Critical Path: 0.01s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
```

### CODE REVIEW

**Comment by mcr229** [comment]
cc @alankelly @gonnet

File: build_srcs.bzl [hunk 1]
```diff
@@ -0,0 +1,146 @@
+# Copyright 2020 Google LLC
```
Comment (gonnet) [inline build_srcs.bzl:1]: `2024`?

File: build_srcs.bzl [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    "src/operators/resize-bilinear-nhwc.c",
+    "src/operators/rope-nthc.c",
+    "src/operators/scaled-dot-product-attention-nhtc.c",
+    "src/operators/slice-nd.c",
+    "src/operators/softmax-nc.c",
+    "src/operators/transpose-nd.c",
+    "src/operators/unary-elementwise-nc.c",
+    "src/operators/unpooling-nhwc.c",
+]
+
```
Comment (gonnet) [inline build_srcs.bzl:39]: Remove extra newline?

**Review by gonnet — APPROVED** [review]
Just some minor nits :slightly_smiling_face:

**Comment by mcr229** [comment]
Thanks @gonnet! Resolved them!

**Pull request merged**